network.s.dfy

```diff
@@ -27,14 +27,19 @@ module HostIdentifiers {
 // the Message type before the Network module. (Contrast with ch05/ex02.)
 module Network {
   import opened Library
+  import opened HostIdentifiers
+
+  datatype Message<Payload(==)> = Message(sender:HostId, payload:Payload)
 
   // A MessageOps is a "binding variable" used to connect a Host's Next step
   // (what message got received, what got sent?) with the Network (only allow
   // receipt of messages sent prior; record newly-sent messages).
   // Note that both fields are Option. A step predicate can say recv.None?
   // to indicate that it doesn't need to receive a message to occur.
   // It can say send.None? to indicate that it doesn't want to transmit a message.
-  datatype MessageOps<Message> = MessageOps(recv:Option<Message>, send:Option<Message>) {
+  datatype MessageOps<Payload> = MessageOps(recv:Option<Message<Payload>>, 
+                                            send:Option<Message<Payload>>,
+                                            signatureChecks:set<Message<Payload>>) {
```

Comment:
How about calling this `signedMsgs` rather than `signatureChecks`?

Reply:
Yes, this is a better name, should we retain the hint to the user, might be `signedMsgsToCheck`?